File: app/src/main/java/com/lucas/omnia/adapters/FrontAdapter.java
package com.lucas.omnia.adapters;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.lucas.omnia.R;
import com.lucas.omnia.models.Front;

import java.util.List;

public class FrontAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private final List<Front> frontList;

    public static class FrontViewHolder extends RecyclerView.ViewHolder {
        TextView nameTextView;

        FrontViewHolder(View itemView) {
            super(itemView);

            nameTextView = itemView.findViewById(R.id.front_tv_name_body);
        }
    }

    public FrontAdapter(List<Front> frontList) {
        this.frontList = frontList;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        return new FrontViewHolder(inflater.inflate(R.layout.item_front, parent, false));
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder viewHolder, int position) {
        Front front = frontList.get(position);
        TextView nameTextView = ((FrontViewHolder) viewHolder).nameTextView;

        nameTextView.setText(front.getName());
    }

    @Override
    public int getItemCount() {
        return frontList.size();
    }
}

File: app/src/main/java/com/lucas/omnia/adapters/FrontAdapter.kt
package com.lucas.omnia.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.lucas.omnia.R
import com.lucas.omnia.models.Front

class FrontAdapter(private val frontList: List<Front>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    class FrontViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var nameTextView: TextView = itemView.findViewById(R.id.front_tv_name_body)

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return FrontViewHolder(inflater.inflate(R.layout.item_front, parent, false))
    }

    override fun onBindViewHolder(viewHolder: RecyclerView.ViewHolder, position: Int) {
        val (name) = frontList[position]
        val nameTextView = (viewHolder as FrontViewHolder).nameTextView
        nameTextView.text = name
    }

    override fun getItemCount(): Int {
        return frontList.size
    }
}